ipa-core/src/config.rs

```diff
@@ -131,6 +131,13 @@ impl NetworkConfig<Shard> {
             identities,
         }
     }
+
+    /// # Panics
+    /// In the unlikely event a usize cannot be turned into a u32
```

Comment:
nit: I wouldn't say it is unlikely for `usize`, but it is surely unexpected to have 4B shards